src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -252,6 +258,31 @@ public boolean visible(String name, Map<String, Object> config) {
     }
   }
 
+  /**
+   * A recommender for table names that caches the table names for the last configuration.
+   */
+  private static class CachingTableRecommender extends TableRecommender {
+
+    private Map<String, Object> lastConfig;
+    private List<Object> results;
+
+    @Override
+    public List<Object> validValues(String name, Map<String, Object> config) {
+      if (lastConfig != null && lastConfig.equals(config)) {
```

Comment:
Okay, that makes sense. I can add a short timeout and make it threadsafe, but would it also be worthwhile to change the `JdbcSourceConnector.config()` method to not reuse the static instance? That's currently the only place the static `CONFIG_DEF` instance is currently used.